scripts/test-check-packages.sh

```diff
@@ -14,10 +14,14 @@ elastic-package stack up -d -v
 
 eval "$(elastic-package stack shellinit)"
 
+echo "***** DEBUGGING *****"
+echo $WORKSPACE
+echo "***** DEBUGGING *****"
+
 for d in test/packages/*/; do
   (
     cd $d
     elastic-package check -v
-    elastic-package test -v --report-format xUnit --report-output file
+    elastic-package test -v --report-format xUnit --report-output /tmp/build/test-results/
```

Comment:
Keep in mind that these tests should be also possible to run using `make check` locally.

Reply:
Good point. I guess we _do_ have a case for a default location (related to the discussion in https://github.com/elastic/elastic-package/pull/158#issuecomment-720338982). I will revert the recent changes in this PR so the `--report-output` flag once again takes either `stdout` or `file`. And if it's `file`, it will simply write to the build folder determined by calling `FindBuildPackagesDirectory()`.

Reply:
```
REPORT_FOLDER=$(pwd)/build/test-results/
mkdir -p ${REPORT_FOLDER}

for d in test/packages/*/; do

   ...
  elastic-package test -v --report-format xUnit --report-output "${REPORT_FOLDER}"
```

Reply:
Thanks @v1v. 

Chatting with @mtojek about this off-PR I realized that the build directory location is tightly coupled to the `elastic-package` tool. That is, it's baked in to the tool's code itself and not configurable from the outside. Putting aside whether this is good or bad, for now it makes sense to stay consistent with this model. 

As a result I'm going to update this PR again so that `--report-output` cannot take a path to the build directory. Instead it will take `file` as a valid value (as it used to before this PR), and if that value is detected, it will write the test results to the build directory that's baked into `elastic-package`.